src/Corax/IndexSearcher/MultiTermMatches/IndexSearcher.MultiTermMatches.In.cs

```diff
@@ -137,70 +137,70 @@ public IQueryMatch AllInQuery(FieldMetadata field, HashSet<string> allInTerms)
         return UnaryQuery(stack[0], field, list, UnaryMatchOperation.AllIn, -1);
     }
 
-    public MultiTermMatch InQuery<TScoreFunction>(FieldMetadata binding, List<string> inTerms, TScoreFunction scoreFunction)
-        where TScoreFunction : IQueryScoreFunction
-    {
-        var terms = _fieldsTree?.CompactTreeFor(binding.FieldName);
-        if (terms == null)
-        {
-            // If either the term or the field does not exist the request will be empty. 
-            return MultiTermMatch.CreateEmpty(_transaction.Allocator);
-        }
-
-        if (inTerms.Count is > 1 and <= 4)
-        {
-            var stack = new BinaryMatch[inTerms.Count / 2];
-            for (int i = 0; i < inTerms.Count / 2; i++)
-            {
-                var term1 = Boost(TermQuery(binding, inTerms[i * 2], terms), scoreFunction);
-                var term2 = Boost(TermQuery(binding, inTerms[i * 2 + 1], terms), scoreFunction);
-                stack[i] = Or(term1, term2);
-            }
-
-            if (inTerms.Count % 2 == 1)
-            {
-                // We need even values to make the last work. 
-                var term = Boost(TermQuery(binding, inTerms[^1], terms), scoreFunction);
-                stack[^1] = Or(stack[^1], term);
-            }
-
-            int currentTerms = stack.Length;
-            while (currentTerms > 1)
-            {
-                int termsToProcess = currentTerms / 2;
-                int excessTerms = currentTerms % 2;
-
-                for (int i = 0; i < termsToProcess; i++)
-                    stack[i] = Or(stack[i * 2], stack[i * 2 + 1]);
-
-                if (excessTerms != 0)
-                    stack[termsToProcess - 1] = Or(stack[termsToProcess - 1], stack[currentTerms - 1]);
-
-                currentTerms = termsToProcess;
-            }
-
-            return MultiTermMatch.Create(stack[0]);
-        }
-
-        return MultiTermMatch.Create(
-            MultiTermBoostingMatch<InTermProvider>.Create(
-                this, new InTermProvider(this, binding, inTerms), scoreFunction));
-    }
+    // public MultiTermMatch InQuery<TScoreFunction>(FieldMetadata field, List<string> inTerms, TScoreFunction scoreFunction)
```

Comment:
Deleted. Since `ConnstantScoreFunc` is now a parameter in the equation I decided to change the formula a little bit:
`BoostingMatch` is a holder for 'scalar' now. (and it will pass it in `Score` call) There is no reason to have two implementations of most primitives (boosted and nonboosted).